scp.py

```diff
@@ -177,13 +177,21 @@ def putfo(self, fl, remote_path, mode='0644'):
         @type remote_path: str
         @param mode: permissions (posix-style) for the uploaded file
         @type mode: str
+        @param length: length of the file in bytes. If ``None``, the length
+            will be computed using `seek()` and `tell()`.
         """
+        if length is None:
+            pos = fl.tell()
+            fl.seek(0, 2)  # Seek to end
```

Comment:
Could ``os.SEEK_END`` be used instead of ``2`` here? It would improve readability, and ``os`` is already imported anyway.